test/integration/provider/rti.go

```diff
@@ -0,0 +1,90 @@
+package provider
+
+import (
+	"fmt"
+
+	gcp "github.com/gardener/machine-controller-manager-provider-gcp/pkg/gcp"
+	v1alpha1 "github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	// IntegrationTestTag is specifically used for integration test
+	// in case the test is run against non seed cluster then the supplied MachineClass
+	// is expected to have these tags set so that the machines created from this suite
+	// won't be orphan collected.
+	IntegrationTestTag = "kubernetes-io-role-integration-test"
+)
+
+//ResourcesTrackerImpl type keeps a note of resources which are initialized in MCM IT suite and are used in provider IT
+type ResourcesTrackerImpl struct {
+	MachineClass *v1alpha1.MachineClass
+	SecretData   map[string][]byte
+	ClusterName  string
+}
+
+//InitializeResourcesTracker initializes the type ResourcesTrackerImpl variable and tries
+//to delete the orphan resources present before the actual IT runs.
+func (r *ResourcesTrackerImpl) InitializeResourcesTracker(machineClass *v1alpha1.MachineClass, secretData map[string][]byte, clusterName string) error {
+
+	r.MachineClass = machineClass
+	r.SecretData = secretData
+	r.ClusterName = clusterName
+
+	initialVMs, initialVolumes, initialMachines, err := r.probeResources()
+
+	if err != nil {
+		fmt.Printf("Error in initial probe of orphaned resources: %s", err.Error())
+		return err
+	}
+
+	if initialVMs != nil || initialVolumes != nil || initialMachines != nil {
+		err := fmt.Errorf("orphan resources are available. Clean them up before proceeding with the test.\nvirtual machines: %v\nvolumes: %v\nmcm machines: %v", initialVMs, initialVolumes, initialMachines)
+		return err
+	}
+	return nil
+}
+
+// probeResources will look for orphaned resources and returns
+// those resources which could not be deleted in the order
+// orphanedInstances, orphanedVolumes, orphanedMachines
+func (r *ResourcesTrackerImpl) probeResources() ([]string, []string, []string, error) {
+
+	ms := gcp.NewGCPPlugin(&gcp.PluginSPIImpl{})
+	ctx, svc, err := ms.SPI.NewComputeService(&corev1.Secret{Data: r.SecretData})
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	orphanedInstances, err := getOrphanedVMs(ctx, svc, IntegrationTestTag, r.MachineClass, r.SecretData)
```

Comment:
`IntegrationTestTag` is already a global variable. This need not be passed as parameters to functions. It will be accessible directly. This change has to be made in IT suite of other providers as well.

Reply:
Yes I thought so too, but keeping the tag there make the function more readable , means one could tell by the function call only that we are looking for orphaed VMs with a certain tag. If we remove the tag, and keep the same fn name `getOrphanedVM` then it could be misleading, thats why I kept it there. WDYT?
and in case of provider-aws we will be making two fns misleading because there both `getOrphanedVM` and `getOrphanedVolumes` accept tags as parameters..

Reply:
>If we remove the tag, and keep the same fn name getOrphanedVM then it could be misleading

Don't worry. **_It won't be misleading_**. It is obvious as well as a convention that virtual machines created for the purpose of integration tests will have distinct identifiable tags (which we can also mention in the documentation). This is also one of the reasons why the variable `IntegrationTestTag` is made global. So that, if the value changes in the future then there is only a single line of code to be changed. 

As far as readability is concerned, you could add a comment for the functions employing this global variable to provide more clarity. But, passing a global variable as a parameter to a function, straight away conflicts the definition of a global variable.